torch/csrc/jit/codegen/cuda/kernel_ir.h

```diff
@@ -1244,9 +1269,12 @@ class TORCH_CUDA_CU_API ForLoop final : public Expr {
 
  private:
   Val* const index_ = nullptr;
+  Val* const extent_ = nullptr;
   IterDomain* const iter_domain_;
   Scope body_;
   bool unroll_ = false;
+  bool vectorize_ = false;
+  Val* shift_ = nullptr;
```

Comment:
Is `shift_` supposed to be used only when `vectorize` is true?

Reply:
`[pre | vectorize | post]`  <= inner-most, merged root domain
`shift_` is applied to the `vectorize` and `post` sections.

Reply:
When is `vectorize_` true? Please comment them.

Reply:
// vectorize is true when the for-loop contains a vectorize set
  // the flag is used to omit the for-loop from the kernel